src/crawlee/_request.py

```diff
@@ -119,7 +120,12 @@ class BaseRequestData(BaseModel):
     method: HttpMethod = 'GET'
     """HTTP request method."""
 
-    headers: Annotated[dict[str, str], Field(default_factory=dict)] = {}
+    headers: Annotated[
+        HttpHeaders,
+        # Normalize headers to lowercase keys and sort them.
+        PlainValidator(lambda value: normalize_headers(value)),
+        Field(default_factory={}),
```

Comment:
Shouldn't `default_factory` be a callable? And do we really need it?

Reply:
oh, good catch